### ISSUE TITLE
[Impressions] About RC

### ISSUE THREAD

**Issue by nonunknown** [radubolovan/Godot-Database-Manager#5]
Here I'll leave my impressions with the new version of GDM since I'm gonna use it on my ASM (Abstract State Machine) to store useful data.

* I've downloaded from branch 2.0 (in-editor it shows 1.0 RC, mistake?) 
* Really loved the new floating window system
* Maybe a shortcut for closing/opening the GDM window? like F5. Or a button on main windows in the side of run scene buttons
![details](https://user-images.githubusercontent.com/58845030/77579349-ea2bd380-6eb8-11ea-9fcc-2d387758debc.png)

**I - Tabs can not be closed**
   - Add a X icon to close
   - Add a CTRL-W shortcut to close

**II - Icons need adjustment**
   - Too big? (aesthetic Lol)
   - Needs tooltips example: leave the mouse above the X icon shows a box saying: delete table

**III - Most Important TYPES**
   - Why I should use GDM instead of Resource saving? Simple, You can save ANY Type (at least this is why I want to use GDM)
   - So I suddenly realized I cant use GDM as it is, cuz I cant save my custom types and other built-in types like Vector2,Vector3.
   - For example if I have A Custom class called Connections:
```
class_name Connections
var from:GraphNode
var to:GraphNode
var from_clicked:Vector2
var to_clicked:Vector2
```
   - what if I want to save this class in GDM? maybe users have to create a serialize function?
``` 
# continuing the class mentioned above
func serialize() -> Dictionary:
   var dict
   dict["from"] = from 
   dict["to"] = to
   dict["from_clicked"] = from_clicked
........
   return dict
```
 - Implementing [basic built in types](https://docs.godotengine.org/en/3.1/getting_started/scripting/gdscript/gdscript_basics.html#basic-built-in-types) could solve the problem since whe can convert a class to a dictionary!
**IV - Properties Aesthetic**
This is more about aesthetic, I think that showing the ID of a property to the end-user causes confusion (maybe I have TOC?)

**Comment by radubolovan** [comment]
I - Tabs can not be closed: Impossible. There is a bug in Godot Engine. Waiting for them to fix it: https://github.com/godotengine/godot/issues/33649

II - Icons need adjustment : not a priority, but yes they should look better. Plan for 3.0

III - This is a plan for 3.0

IV - I don't understand the problem. Currently the ID of the properties are hidden from the user interface. That ID that you see is a property in the table with the name "ID".

**Comment by radubolovan** [comment]
I - Tabs can not be closed: Impossible. There is a bug in Godot Engine. Waiting for them to fix it: godotengine/godot#33649

II - Icons need adjustment : not a priority, but yes they should look better. Plan for 3.0

III - This is a plan for 3.0

IV - I think you are right. I'll hide them for now.

**Comment by nonunknown** [comment]
@radubolovan Nice, thanks! Please take a look at #6 . I'll try to do a tab closing system and do another pull request.

**Comment by nonunknown** [comment]
@radubolovan I'll not do a PR for this, cuz maybe is a dirty trick and you dont want to use it, but I've placed the following code at the:
**db_interface.gd**
```
func _input(event):
	if event is InputEventKey:
		if event.is_pressed() and event.scancode == KEY_W:
			var tab = $dlg/databases.get_current_tab_control()
			$dlg/databases.call_deferred("remove_child",tab)
			pass
```
worked fine, I was thinking about CTRL+W for that

**Comment by radubolovan** [comment]
Carreful with the keyboard shortcuts!
I tried to use CTRL+S to save current database in the past and I dropped the feature.
_input function gets called even if the plugin's window / dialog is closed. Add a print (test) in there and you'll see.

**Comment by nonunknown** [comment]
for the open/close window running when is closed is ok, but just making a bool checking if the window is active.

**Comment by radubolovan** [comment]
Yes, that could work: simply return in the beginning of the _input function if the window is not active.

**Comment by radubolovan** [comment]
I have completed implementing all the features for 2.0.
Today I'm starting to make some how_to/documentation/examples and hopefully I'll release 2.0 alpha next week.

By the way, this is how I want to proceed with developing the current version and the next ones:
- pre_alpha:
1. make a plan for the current version
2. create a separated branch on github

- alpha:
1. all features implemented
2. all known bugs fixed

- beta:
1. find and fix all remaining bugs (or as many as possible)

- release candidate:
1. test the product in production and fix all remaining bugs

- release:
1. merge the current branch into master
2. publish the release the version
3. make a request on the Godot Asset Library

**Comment by nonunknown** [comment]
OK, for better testing purposes, plz launch the docs with the pre-alpha

**Comment by radubolovan** [comment]
Hey there,

I added most of the documentation. Still WIP though.
I also fixed a few issues that I found lately and added a few more features.

If you fetch / pull, I think this one will be in your interest:
https://github.com/radubolovan/Godot-Database-Manager/issues/9

If you have problems converting your database to the new format, just edit the JSON file and add this right after the first "{":
"GDDB_ver": "1.0",

For more details, see the database example:
https://github.com/radubolovan/Godot-Database-Manager/blob/2.0/prj/game_data.json
